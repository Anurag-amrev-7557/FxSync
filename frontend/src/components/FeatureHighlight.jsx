import React, { useId } from 'react';
import PropTypes from 'prop-types';
import { SparkleIcon } from './Icons';

const FeatureHighlight = ({ icon, text, tooltip, accent, delay = 0, tabIndex = 0, ariaLabel, ...props }) => {
  const tooltipId = useId();
  return (
    <div
      className={`relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105`}
      style={{ transitionDelay: `${delay}ms` }}
      tabIndex={tabIndex}
      role="button"
      aria-label={ariaLabel || text}
      aria-describedby={tooltipId}
      {...props}
    >
      <div
        className={`flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br ${accent} transition-all duration-300 shadow-md`}
      >
        {icon}
      </div>
      <span className="transition-all duration-300 group-hover:translate-x-1 font-semibold">
        {text}
      </span>
      {/* Tooltip on hover/focus */}
      <div
        id={tooltipId}
        className="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
        role="tooltip"
        aria-live="polite"
      >
        <span className="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap">
          {tooltip}
        </span>
      </div>
      {/* Accent sparkle */}
      <SparkleIcon className="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle" />
    </div>
  );
};

FeatureHighlight.propTypes = {
  icon: PropTypes.node.isRequired,
  text: PropTypes.string.isRequired,
  tooltip: PropTypes.string.isRequired,
  accent: PropTypes.string.isRequired,
  delay: PropTypes.number,
  tabIndex: PropTypes.number,
  ariaLabel: PropTypes.string,
};

export default FeatureHighlight; 