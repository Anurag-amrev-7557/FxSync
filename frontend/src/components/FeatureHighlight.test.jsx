import React from 'react';
import { render, screen } from '@testing-library/react';
import FeatureHighlight from './FeatureHighlight';

describe('FeatureHighlight', () => {
  const icon = <svg data-testid="icon" />;
  const text = 'Test Feature';
  const tooltip = 'Test tooltip';
  const accent = 'from-blue-400/30 to-blue-200/10';

  it('renders and matches snapshot', () => {
    const { asFragment } = render(
      <FeatureHighlight icon={icon} text={text} tooltip={tooltip} accent={accent} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders icon, text, and tooltip', () => {
    render(<FeatureHighlight icon={icon} text={text} tooltip={tooltip} accent={accent} />);
    expect(screen.getByTestId('icon')).toBeInTheDocument();
    expect(screen.getByText(text)).toBeInTheDocument();
    expect(screen.getByRole('tooltip')).toHaveTextContent(tooltip);
  });

  it('is accessible as a button with aria-label and tooltip', () => {
    render(
      <FeatureHighlight icon={icon} text={text} tooltip={tooltip} accent={accent} ariaLabel="Custom label" />
    );
    const button = screen.getByRole('button', { name: /custom label/i });
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('aria-describedby');
    const tooltipEl = screen.getByRole('tooltip');
    expect(tooltipEl).toHaveTextContent(tooltip);
  });
}); 