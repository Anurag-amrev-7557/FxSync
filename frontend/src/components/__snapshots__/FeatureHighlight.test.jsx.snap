// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FeatureHighlight renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-describedby=":r0:"
    aria-label="Test Feature"
    class="relative flex items-center gap-2 text-xs sm:text-sm text-neutral-400 group cursor-pointer transition-all duration-300 hover:text-white hover:scale-105"
    role="button"
    style="transition-delay: 0ms;"
    tabindex="0"
  >
    <div
      class="flex items-center justify-center w-8 h-8 rounded-full bg-gradient-to-br from-blue-400/30 to-blue-200/10 transition-all duration-300 shadow-md"
    >
      <svg
        data-testid="icon"
      />
    </div>
    <span
      class="transition-all duration-300 group-hover:translate-x-1 font-semibold"
    >
      Test Feature
    </span>
    <div
      aria-live="polite"
      class="absolute left-1/2 -translate-x-1/2 top-full mt-2 z-20 opacity-0 group-hover:opacity-100 group-focus:opacity-100 pointer-events-none transition-opacity duration-300"
      id=":r0:"
      role="tooltip"
    >
      <span
        class="px-2 py-1 rounded bg-neutral-900/90 text-xs text-neutral-200 shadow-lg border border-neutral-700 whitespace-nowrap"
      >
        Test tooltip
      </span>
    </div>
    <svg
      class="absolute -top-2 -right-2 w-3 h-3 text-white opacity-0 group-hover:opacity-80 transition-all duration-500 group-hover:animate-twinkle"
      fill="currentColor"
      viewBox="0 0 20 20"
    >
      <polygon
        points="10,2 12,8 18,8 13,12 15,18 10,14 5,18 7,12 2,8 8,8"
      />
    </svg>
  </div>
</DocumentFragment>
`;
